searx/webapp.py

```diff
@@ -297,6 +298,24 @@ def morty_proxify(url: str):
     return '{0}?{1}'.format(settings['result_proxy']['url'], urlencode(url_params))
 
 
+def favicon_proxify(url: str):
+    # url is a FQDN (e.g. example.com, en.wikipedia.org)
+
+    resolver = request.preferences.get_value('favicon_resolver')
+
+    # if resolver is empty, just return nothing
+    if not resolver:
+        return ""
+
+    # check resolver is valid
+    if resolver not in favicon_backends:
+        return ""
+
+    h = new_hmac(settings['server']['secret_key'], url.encode())
+
+    return '{0}?{1}'.format(url_for('favicon_proxy'), urlencode(dict(q=url.encode(), h=h)))
```

Comment:
```suggestion
    return f"{url_for('favicon_proxy')}?{urlencode(dict(q=url.encode(), h=h)))}"
```

Reply:
@Bnyro thanks for your review, but please what with this PR / I'm in review with this PR .. I modified a lot .. I will need some more time to finish my rework .. Then you can make a review.